cypress-io/cypress
# fix: Fix sourcemaps when using cy.origin() dependencies

- Closes #23365

### User facing changelog

- When utilizing dependencies within the `cy.origin()` callback, errors now show the correct code frame

### Additional details

Ensures a new source map is generated when the webpack-preprocessor changes the test code, fixing what the code frame displays.

### Steps to test

Run [this test](https://github.com/cypress-io/cypress/blob/4e667e5383a4df482756f3b1b3d572c3a97ac7df/packages/driver/cypress/e2e/e2e/origin/dependencies.cy.ts#L7-L13
), but change the assertion to fail:

```js
  it('works with require()', () => {
    cy.origin('http://www.foobar.com:3500/', () => {
      const lodash = require('lodash')

      expect(lodash.get({ foo: 'foo' }, 'foo')).to.equal('bar')
    })
  })
```

The code frame should point to `cy.origin` and show the original source code.

### How has the user experience changed?

Before

![Image](https://user-images.githubusercontent.com/1157043/184896931-d448a5bc-cd50-479a-bf91-dfe011e73a04.png)

After

![Screen Shot 2022-10-24 at 11 15 20 AM](https://user-images.githubusercontent.com/1157043/197562193-60bfbda7-bdef-4921-91ae-bcfcdf3d2ae9.png)

### PR Tasks


- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [N/A] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
